File: 0001-two-sum/0001-two-sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut seen: HashMap<i32, usize> = HashMap::new();

        for (i, num) in nums.iter().enumerate() {
            let complement = target - num;
            if seen.contains_key(&complement) {
                return vec![seen[&complement] as i32, i as i32];
            }
            seen.insert(*num, i);
        }

        vec![]
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        pair_idx = {}

        for idx, num in enumerate(nums):
            pair_needed = target - num
            if pair_needed in pair_idx:
                return [idx, pair_idx[pair_needed]]
            else:
                pair_idx[num] = idx
        
        return []